lib/SILOptimizer/Utils/CFGOptUtils.cpp

```diff
@@ -700,7 +701,73 @@ void swift::completeJointPostDominanceSet(
 
   // Now that we are done, add all remaining must visit blocks to our result
   // list. These are the remaining parts of our joint post-dominance closure.
-  llvm::copy(mustVisitSuccessorBlocks, std::back_inserter(result));
+  llvm::copy(mustVisitSuccessorBlocks, std::back_inserter(completion));
+
+  // Also, compute the postdominance set, which is the union of the user blocks
+  // that are not visited in the above fix point iteration and the
+  // mustVisitSuccessorBlocks.
+  llvm::copy_if(userBlocks, std::back_inserter(postDominators),
+                [&](SILBasicBlock *block) -> bool {
+                  return !visitedBlocks.count(block);
+                });
+  postDominators.append(completion.begin(), completion.end());
+}
+
+void swift::computePostDominatorsOfUsers(
+    SmallPtrSetImpl<SILInstruction *> &users,
+    SmallPtrSetImpl<SILInstruction *> &defs,
+    SmallVectorImpl<SILInstruction *> &postDominators) {
+  // Step 1. Collect the basic blocks of the users and defs.
+  SmallPtrSet<SILBasicBlock *, 4> userBlocks;
+  for (auto *inst : users) {
+    userBlocks.insert(inst->getParent());
+  }
+  SmallPtrSet<SILBasicBlock *, 4> defBlocks;
+  for (auto *inst : defs) {
+    defBlocks.insert(inst->getParent());
+  }
+
+  // Step 2. Compute post-dominators of all user blocks.
+  SmallVector<SILBasicBlock *, 8> completions;
+  SmallVector<SILBasicBlock *, 8> postDominatingBlocks;
+  completeJointPostDominanceSet(userBlocks, defBlocks, completions,
+                                postDominatingBlocks);
+
+  // Step 3. Compute all post-dominating instructions by looking into the
+  // post-dominating blocks.
+  SmallPtrSet<SILBasicBlock *, 8> completionSet(completions.begin(),
+                                                completions.end());
```

Comment:
I don't think you should copy `completions`. You should be able to check membership in `userBlocks` instead.

Reply:
That's right. Will fix this.